java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java

```diff
@@ -11,17 +11,75 @@
 
 package com.microsoft.java.test.plugin.util;
 
+import com.microsoft.java.test.plugin.model.JavaTestItem;
+import com.microsoft.java.test.plugin.model.TestKind;
+import com.microsoft.java.test.plugin.model.TestLevel;
+
+import org.eclipse.core.runtime.IPath;
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaModelException;
 import org.eclipse.jdt.core.dom.AST;
 import org.eclipse.jdt.core.dom.ASTNode;
 import org.eclipse.jdt.core.dom.ASTParser;
 import org.eclipse.jdt.core.dom.CompilationUnit;
 import org.eclipse.jdt.core.manipulation.CoreASTProvider;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.ProjectUtils;
+import org.eclipse.jdt.ls.core.internal.ResourceUtils;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
 
 @SuppressWarnings("restriction")
 public class TestSearchUtils {
 
+    public static List<JavaTestItem> findJavaProjects(List<Object> arguments, IProgressMonitor monitor) {
+        if (arguments == null || arguments.size() == 0) {
+            return Collections.emptyList();
+        }
+        final String workspaceFolderUri = (String) arguments.get(0);
+        final IPath workspaceFolderPath = ResourceUtils.filePathFromURI(workspaceFolderUri);
+        if (workspaceFolderPath == null) {
+            JUnitPlugin.logError("Failed to parse workspace folder path from uri: " + workspaceFolderUri);
+            // todo: handle non-file scheme
+            return Collections.emptyList();
+        }
+        final List<IJavaProject> javaProjects = new LinkedList<>();
+        for (final IJavaProject project : ProjectUtils.getJavaProjects()) {
```

Comment:
Among `ProjectUtils.getJavaProjects()`, what else are you going to filter together with default project?

Reply:
No other projects so far. All the projects that can be seen in user's workspaces should be listed

Reply:
Just curious about if we can simply remove default project, and skip the "prefix checking"?

Reply:
prefix checking it totally fine with me, just curious about it.

Reply:
Oh, that's a good point. As far as I know, currently we could have default project, visible project and invisible project. It seems that we can just remove the default project and put other Java projects into the list.